### PR TITLE
Reduce code duplication down to 15 lines per piece

### PR DESCRIPTION
**Pull request by c71n93**
Related to #2863

---

## PR-Codex overview
This PR updates Java versions, refactors saving logic, and improves test parameterization.

### Detailed summary
- Updated Java version to 17
- Refactored saving logic in `HmOptional` and `HmBase`
- Improved test parameterization in `UnplaceMojoTest`

> The following files were skipped due to too many changes: `eo-maven-plugin/src/test/java/org/eolang/maven/UnplaceMojoTest.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by yegor256** [comment]
@c71n93 what's up with this one? Still a draft?

**Comment by c71n93** [comment]
@yegor256 still working on it

**Comment by maxonfjvipon** [comment]
@c71n93 when one part of the methods is placed inside abstract class and other part is placed inside concrete class - it's a marker of bad design. If we're not talking about corner cases like mojos, the only right usage of abstract classes is envelope pattern: https://www.yegor256.com/2017/01/31/decorating-envelopes.html

**Comment by maxonfjvipon** [comment]
@c71n93 what's up with this one?

File: eo-runtime/lib/groovy-5.0.0-alpha-1-sources.jar
Comment (maxonfjvipon) [inline eo-runtime/lib/groovy-5.0.0-alpha-1-sources.jar:1]: @c71n93 I don't think these binaries belong here

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]

File: eo-runtime/lib/groovy-5.0.0-alpha-1-sources.jar
Comment (c71n93) [reply eo-runtime/lib/groovy-5.0.0-alpha-1-sources.jar:1]: @maxonfjvipon you are right)

**Comment by c71n93** [comment]
@maxonfjvipon what do you think about `Home` interface now? Is it acceptable?

**Comment by c71n93** [comment]
@maxonfjvipon all resolved, could you review this one, please?

File: eo-runtime/src/test/java/EOorg/EOeolang/EOtryTest.java [hunk 1]
```diff
@@ -86,25 +86,6 @@ public void usesCatcherOutput() {
         );
     }
 
-    @Test
```
Comment (maxonfjvipon) [inline eo-runtime/src/test/java/EOorg/EOeolang/EOtryTest.java:89]: @c71n93 why was this test removed?

File: eo-maven-plugin/src/main/java/org/eolang/maven/util/DataSaved.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import java.nio.file.Path;
+import org.cactoos.Input;
+import org.cactoos.Text;
+import org.cactoos.io.InputOf;
+
+/**
+ * The data is saved to a file.
+ * @since 0.32.0
+ */
+public interface DataSaved {
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/main/java/org/eolang/maven/util/DataSaved.java:37]: @c71n93 interface with default methods - it's almost the same as abstract class. You split the logic and keep it in different places.

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]

File: eo-runtime/src/test/java/EOorg/EOeolang/EOtryTest.java (hunk 1 above)
Comment (c71n93) [reply eo-runtime/src/test/java/EOorg/EOeolang/EOtryTest.java:89]: there was 2 same tests with different names

**Comment by c71n93** [comment]
@maxonfjvipon could you check this one again, please?

**Review by maxonfjvipon — APPROVED** [review]

**Comment by c71n93** [comment]
@yegor256 could you check this one, please?

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**